src/machine/machine_atmega.go

```diff
@@ -5,11 +5,120 @@ package machine
 
 import (
 	"device/avr"
+	"errors"
 	"runtime/interrupt"
 	"runtime/volatile"
 	"unsafe"
 )
 
+// EECR bitfield values.
+// The constants provided by the AVR package are not correct, so these are the correct fields instead.
+const (
+	eecrEERE  = avr.EECR_EERE - 1
+	eecrEEPE  = avr.EECR_EEPE - 1
+	eecrEEMPE = avr.EECR_EEMPE - 2
+	eecrEERIE = avr.EECR_EERIE - 5
+	eecrEEPM0 = avr.EECR_EEPM0 - 12
+	eecrEEPM1 = avr.EECR_EEPM1 - 27
+)
+
+// EEPROM on ATMega
+type EEPROM struct {
+}
+
+var EEPROM0 = EEPROM{}
+
+// setAddress sets the address for a given read or write into the MCUs EEARH/L register.
+func (e EEPROM) setAddress(addr int64) error {
```

Comment:
This is an unexported function, so it's best to use the most efficient data type available. In this case, `int16` (or even `uint16`).

```suggestion
func (e EEPROM) setAddress(addr int16) error {
```

Reply:
If I make this change, within this function there is a comparison between an `int64` and `int16`. Should I be casting the `int16` into an `int64` to perform this check, or should I change the result of the `Size()` function to return `int16` instead?

Reply:
In this case you know it will never be bigger than an `int16` so casting to the most efficient type is best (for AVR that is `int16`).